websites/T/Trakt/presence.ts

```diff
@@ -0,0 +1,169 @@
+		else if (path.includes("/network" || "/network/friends"))
+			presenceData.details = `Browsing ${username}'s friends`;
+		else presenceData.details = `Browsing ${username}'s profile`;
+	} else if (path.includes("/shows/")) {
+		if (path.includes("/episodes/")) {
+			presenceData.details = "Browsing episode";
+			presenceData.state = `${
+				document
+					.querySelector(
+						"#summary-wrapper > div.container.summary > div > div > div.col-md-10.col-md-offset-2.col-sm-9.col-sm-offset-3.mobile-title > h2 > a:nth-child(1)"
+					)
+					.textContent.split(":")[0]
+			} -
+		${
+			document.querySelector(
+				"#summary-wrapper > div.container.summary > div > div > div.col-md-10.col-md-offset-2.col-sm-9.col-sm-offset-3.mobile-title > h1 > span.main-title-sxe"
+			).textContent
+		}`;
+			presenceData.buttons = [{ label: "Check episode", url: document.URL }];
+		} else if (path.includes("/seasons/")) {
+			presenceData.details = "Browsing season";
+			presenceData.state = `${
+				document.querySelector("#level-up-link").textContent
+			}: ${
+				document.querySelector(
+					"#summary-wrapper > div.container.summary > div > div > div.col-md-10.col-md-offset-2.col-sm-9.col-sm-offset-3.mobile-title > h1"
+				).firstChild.textContent
+			}`;
+			presenceData.buttons = [{ label: "Check season", url: document.URL }];
+		} else {
+			try {
+				presenceData.details = "Browsing TV show";
+				presenceData.state = document.querySelector(
+					"#summary-wrapper > div.container.summary > div > div > div.col-md-10.col-md-offset-2.col-sm-9.col-sm-offset-3.mobile-title > h1"
+				).firstChild.textContent;
+				presenceData.buttons = [{ label: "Check TV show", url: document.URL }];
+			} catch (e) {
+				presenceData.details = "Browsing";
+			}
+		}
+	} else if (path.includes("/movies")) {
+		try {
+			presenceData.details = "Browsing movie";
+			presenceData.state = document
+				.querySelector(
+					"#summary-wrapper > div.container.summary > div > div > div.col-md-10.col-md-offset-2.col-sm-9.col-sm-offset-3.mobile-title > h1"
+				)
+				.textContent.split(
+					document.querySelector(
+						"#summary-wrapper > div.container.summary > div > div > div.col-md-10.col-md-offset-2.col-sm-9.col-sm-offset-3.mobile-title > h1 > span"
+					).textContent
+				)[0];
+			presenceData.buttons = [{ label: "Check movie", url: document.URL }];
+		} catch (e) {
+			presenceData.details = "Browsing";
+		}
+	} else if (path.includes("/people")) {
+		presenceData.details = "Browsing actor";
+		presenceData.state = document.querySelector(
+			"#summary-wrapper > div.container.summary > div > div > div.col-md-10.col-md-offset-2.col-sm-9.col-sm-offset-3.mobile-title > h1"
+		).textContent;
+		presenceData.buttons = [{ label: "Check actor", url: document.URL }];
+	} else if (path.includes("/search")) {
+		presenceData.details = "Searching for";
+		presenceData.state = new URLSearchParams(window.location.search).get(
+			"query"
+		);
+	}
```

Comment:
this entire chain of ifs can be turned into a switch statement

Reply:
how? this if contains `path.includes`, how can i do it with switch?

Reply:
```js
switch (true) {
	case pathname.includes("abc"): ...
}```
but if you don't wanna do this it's fine, `switch true` cases look a bit messy

Reply:
i think if is better way, but if you want, i can change it

Reply:
From what I have tested switch is 10 times slower
![image](https://user-images.githubusercontent.com/68614968/174448560-d5b4f86b-df80-4774-9e5b-27657a4ff3d7.png)